apps/wallet-mobile/src/TxHistory/TxHistory.tsx

```
@@ -36,7 +37,10 @@ export const TxHistory = () => {
     setActiveTab(tab)
   }
 
-  const {sync, isLoading} = useSync(wallet)
+  const {sync, isLoading: isLoadingWallet} = useSync(wallet)
+  const {isLoading: isLoadingPoolTransition} = usePoolTransitionModal()
```

since is requested while opening the wallet, it should swallow errors, otherwise a failed request won't allow the user to open the wallet, it will happen again when opening so it looks safe.

---

it can also leverage the error boundary with a retry

---

Not sure I understand. If usePoolTransitionModal got an error from usePoolTransition, isLoading should still go to false.

---

what will happen to user if getPoolTrasitino fails?

---

It's inside useQuery, so you'll get an error returned there. Since it's not used, then nothing out of the ordinary happens.

---

got it, so that's fine as long as it doesn't become a suspense.